apps/wallet/src/ui/app/components/HideShowDisplayBox.tsx

```diff
@@ -2,49 +2,98 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { Copy16, EyeClose16, EyeOpen16 } from '@mysten/icons';
-import { useState } from 'react';
+import { cx } from 'class-variance-authority';
+import { useEffect, useState } from 'react';
 
 import { useCopyToClipboard } from '../hooks/useCopyToClipboard';
 import { Link } from '../shared/Link';
 import { Text } from '../shared/text';
 
+const AUTO_HIDE_INTERVAL = 3 * 60 * 1000;
+
 export type HideShowDisplayBoxProps = {
-    value: string;
+    value: string | string[];
+    hideCopy?: boolean;
     copiedMessage?: string;
 };
 
 export function HideShowDisplayBox({
     value,
+    hideCopy = false,
     copiedMessage,
 }: HideShowDisplayBoxProps) {
     const [valueHidden, setValueHidden] = useState(true);
-    const copyCallback = useCopyToClipboard(value, {
-        copySuccessMessage: copiedMessage,
-    });
+    const copyCallback = useCopyToClipboard(
```

Comment:
nit: I'm not the biggest fan of nested ternaries, something like this might be a little more readable

```
const clipboardValue = typeof value === 'string' ? value : value.join(' ');
const copyCallback = useCopyToClipboard(hideCopy ? '' : clipboardValue, ...)
```